sharpd/sharp_zebra.c

```diff
@@ -376,14 +408,20 @@ void route_add(const struct prefix *p, vrf_id_t vrf_id,
 	SET_FLAG(api.flags, ZEBRA_FLAG_ALLOW_RECURSION);
 	SET_FLAG(api.message, ZAPI_MESSAGE_NEXTHOP);
 
-	for (ALL_NEXTHOPS_PTR(nhg, nh)) {
-		api_nh = &api.nexthops[i];
+	/* Only send via ID if nhgroup has been successfully installed */
+	if (nhgid && sharp_nhgroup_id_is_installed(nhgid)) {
```

Comment:
mentioned this below too: this will, if I'm reading it correctly, prevent sharpd from being used to exercise backup nexthops -because zebra will unconditionally report that the group is "installed".

Reply:
yea forgot to do that. Will add.